scionlab/models/user_as.py

```diff
@@ -192,100 +127,208 @@ class Meta:
+            iface_client.update(public_ip=vpn_client.ip,
+                                public_port=att_conf.public_port,
+                                bind_ip=None,
+                                bind_port=None)
+        else:
+            border_router = BorderRouter.objects.first_or_create(self.host)
+            iface_client = Interface.objects.create(border_router=border_router,
+                                                    public_ip=vpn_client.ip,
+                                                    public_port=att_conf.public_port)
+            iface_ap = Interface.objects.create(
+                border_router=att_conf.attachment_point.get_border_router_for_useras_interface(),
+                public_ip=att_conf.attachment_point.vpn.server_vpn_ip)
+        return iface_ap, iface_client
+
     def update(self,
                label,
-               attachment_point,
-               use_vpn,
-               public_ip,
-               public_port,
-               bind_ip,
-               bind_port,
                installation_type):
         """
         Update this UserAS instance and immediately `save`.
         Updates the related host, interface and link instances and will trigger
         a configuration bump for the hosts of the affected attachment point(s).
         """
-        prev_ap = self.attachment_point
-
-        host = self.hosts.get()   # UserAS always has only one host
-
-        if self.isd != attachment_point.AS.isd:
-            self._change_isd(attachment_point.AS.isd)
+        # XXX(andrea_tulimiero): Useless since the model instance has already got the updated fields
+        self.label = label
+        self.host.update()
 
-        host.update(
-            public_ip=public_ip,
-            bind_ip=_VAGRANT_VM_LOCAL_IP if installation_type == UserAS.VM else bind_ip,
-        )
-
-        link = self._get_ap_link()
-        interface_ap = link.interfaceA
-        interface_user = link.interfaceB
-        if use_vpn:
-            vpn_client = self._create_or_activate_vpn_client(attachment_point.vpn)
-            interface_user.update(
-                public_ip=vpn_client.ip,
-                public_port=public_port,
-                bind_port=None
-            )
-            interface_ap.update(
-                border_router=attachment_point.get_border_router_for_useras_interface(),
-                public_ip=attachment_point.vpn.server_vpn_ip,
-                public_port=None,
-            )
-        else:
-            host.vpn_clients.update(active=False)   # deactivate all vpn clients
-            interface_user.update(
-                public_ip=None,
-                public_port=public_port,
-                bind_ip=None,
-                bind_port=bind_port
-            )
-            interface_ap.update(
-                border_router=attachment_point.get_border_router_for_useras_interface(),
-                public_ip=None,
-                public_port=None,
-            )
```

Comment:
As discussed: we'd really like to keep this functionality of just changing the selected attachment point. This is how this was previously implemented; it should be fairly straight forward to adapt this.

Sorry again for not explicitly specifying this and not catching this earlier! Now it all makes sense, the way you've restructured the tests, i just didn't get it :)